saleor/graphql/shop/mutations.py

```diff
@@ -288,3 +288,96 @@ def perform_mutation(cls, _root, info, key_type):
 
         instance.delete()
         return AuthorizationKeyDelete(authorization_key=instance, shop=Shop())
+
+
+class StaffNotificationRecipientInput(graphene.InputObjectType):
+    user = graphene.ID(
+        required=False,
+        description="The ID of the user subscribed to email notifications..",
+    )
+    email = graphene.String(
+        required=False,
+        description="Email address of a user subscribed to email notifications.",
+    )
+    active = graphene.Boolean(
+        required=False, description="Determines if a notification active."
+    )
+
+
+class StaffNotificationRecipientCreate(ModelMutation):
+    class Arguments:
+        input = StaffNotificationRecipientInput(
+            required=True,
+            description="Fields required to create a staff notification recipient.",
+        )
+
+    class Meta:
+        description = "Creates a new staff notification recipient."
+        model = account_models.StaffNotificationRecipient
+        permissions = ("site.manage_settings",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        email = cleaned_input.get("email")
+        user = cleaned_input.get("user")
+        cls.validate_input(instance, email, user)
+        if email:
+            staff_user = account_models.User.objects.filter(email=email).first()
+            if staff_user:
+                cleaned_input["user"] = staff_user
+                cleaned_input["email"] = None
+        return cleaned_input
+
+    @staticmethod
+    def validate_input(instance, email, user):
+        if not instance.id and not email and not user:
+            raise ValidationError(
+                {
+                    "staff_notification": ValidationError(
+                        "User or email is required", code=ShopErrorCode.REQUIRED
+                    )
+                }
+            )
+        if user and not user.is_staff:
+            raise ValidationError(
+                {
+                    "user": ValidationError(
+                        "User has to be staff user", code=ShopErrorCode.REQUIRED
```

Comment:
I would say it's `INVALID` error code, since `REQUIRED` is more suitable in a case when a value is required but it wasn't passed.
```suggestion
                        "User has to be staff user", code=ShopErrorCode.INVALID
```

Reply:
done